src/controller/buffer-controller.ts

```diff
@@ -1,16 +1,82 @@
 /*
  * Buffer Controller
-*/
+ */
 
 import Event from '../events';
 import EventHandler from '../event-handler';
 import { logger } from '../utils/logger';
 import { ErrorTypes, ErrorDetails } from '../errors';
 import { getMediaSource } from '../utils/mediasource-helper';
 
+import { TrackSet } from '../types/track';
+import { Segment } from '../types/segment';
+
+// Add extension properties to SourceBuffers from the DOM API.
+type ExtendedSourceBuffer = SourceBuffer & {
+  ended: boolean
```

Comment:
I think this needs to be optional, because it initially will be `undefined`

```suggestion
  ended?: boolean
```

Reply:
Good point. This must not have been caught by the TypeScript code that normally catches delayed initialization because the constructor isn't defined by TS.